SwissDataScienceCenter/renku-ui
# Allow editing of cloud storage secret credentials

# Changes

- Allow setting and removing credentials on a data source
- Do not prompt for credentials at session launch, if there are stored credentials

# Screenshots

**View credentials**
<img width="715" alt="image" src="https://github.com/SwissDataScienceCenter/renku-ui/assets/1196411/1423c3b4-928e-465c-841a-6fd3b811d513">

<img width="715" alt="image" src="https://github.com/SwissDataScienceCenter/renku-ui/assets/1196411/6e20d897-994a-4dd4-90fc-252b2837b94f">


**Update credentials**
<img width="715" alt="image" src="https://github.com/SwissDataScienceCenter/renku-ui/assets/1196411/f444fe97-8b2e-41a9-8cd7-f275008a498c">

**Clear credentials**
<img width="715" alt="image" src="https://github.com/SwissDataScienceCenter/renku-ui/assets/1196411/e2c355de-2567-461e-9569-0551c37cabcc">


# Testing

To test, try the following:
- Create a data source that requires credentials
- Use the "Update Credentials" option to set the credentials (they cannot yet be stored at creation time)
- Launch a session and check that you are not prompted to provide credentials
- Clear the credentials for the data source (currently, you also need to delete the entries from the secrets page, but the backend will be changed to this itself)


/deploy renku-notebooks=pitch/save-storage-secrets renku-data-services=pitch/save-storage-credentials-save-secrets
